### PR TITLE
fully implement puppeteer@2.1.1/lib/Coverage.js

### PR DESCRIPTION
**Pull request by Granitosaurus**
Mostly asyncio.gather additions and expanding kwargs + general formatting

### CODE REVIEW

File: pyppeteer/coverage.py [hunk 1]
```diff
@@ -90,17 +92,15 @@ def __init__(self, client: CDPSession) -> None:
         """
         return await self._jsCoverage.stop()
 
-    async def startCSSCoverage(self, options: Dict = None, **kwargs: Any
-                               ) -> None:
+    async def startCSSCoverage(self, resetOnNavigation: bool = True) -> None:
         """Start CSS coverage measurement.
 
-        Available options are:
+        Available kwargs are:
```
Comment (Mattwmaster58) [inline pyppeteer/coverage.py]: ...There's no kwargs here?

File: pyppeteer/coverage.py [hunk 2]
```diff
@@ -134,31 +134,36 @@ def __init__(self, client: CDPSession) -> None:
...
-        await self._client.send('Profiler.startPreciseCoverage',
-                                {'callCount': False, 'detailed': True})
-        await self._client.send('Debugger.enable')
-        await self._client.send('Debugger.setSkipAllPauses', {'skip': True})
+        await asyncio.gather(
+            self._client.send('Profiler.enable'),
+            self._client.send('Profiler.startPreciseCoverage',
+                              {'callCount': False, 'detailed': True}),
+            self._client.send('Debugger.enable'),
+            self._client.send('Debugger.setSkipAllPauses', {'skip': True}),
```
Comment (Mattwmaster58) [inline pyppeteer/coverage.py:164]: Do you know if the order of this matters? It seems like the answer would be no, but if any of the items require order it's probably `Profiler.enable` and `Profiler.startPreciseCoverage`. Basically only the return order of the awaitables is guarenteed, not the sheduling order as I understand it (and I could definitely be wrong).

Comment (Mattwmaster58) [reply pyppeteer/coverage.py:164]: Nevermind. It looks like the CPython implementation guarentee's order (although the spec does not techincally), I think it's pretty safe to rely on.

File: pyppeteer/coverage.py [hunk 3]
```diff
@@ -196,16 +199,19 @@ def _onExecutionContextsCleared(self, event: Dict) -> None:
...
-        await self._client.send('Debugger.disable')
+        result = await asyncio.gather(
+            self._client.send('Profiler.takePreciseCoverage'),
+            self._client.send('Profiler.stopPreciseCoverage'),
+            self._client.send('Profiler.disable'),
+            self._client.send('Debugger.disable'),
+        )
         helper.removeEventListeners(self._eventListeners)
 
         coverage: List = []
```
Comment (Mattwmaster58) [inline pyppeteer/coverage.py]: mypy is smart enough to infer this type 😉 
```suggestion
        coverage = []
```

File: pyppeteer/coverage.py [hunk 4]
```diff
@@ -277,14 +284,16 @@ def _onExecutionContextsCleared(self, event: Dict) -> None:
...
-        await self._client.send('DOM.disable')
+        ruleTrackingResponse = await self._client.send('CSS.stopRuleUsageTracking')
+        await asyncio.gather(
+            self._client.send('CSS.disable'),
+            self._client.send('DOM.disable')
+        )
         helper.removeEventListeners(self._eventListeners)
 
         # aggregate by styleSheetId
         styleSheetIdToCoverage: Dict = {}
```
Comment (Mattwmaster58) [inline pyppeteer/coverage.py]: superflous type annotation
```suggestion
        styleSheetIdToCoverage = {}
```

File: pyppeteer/coverage.py [hunk 5]
```diff
@@ -307,9 +316,14 @@ def _onExecutionContextsCleared(self, event: Dict) -> None:
         return coverage
 
 
-def convertToDisjointRanges(nestedRanges: List[Any]  # noqa: C901
-                            ) -> List[Any]:
-    """Convert ranges."""
+def convertToDisjointRanges(nestedRanges: List[dict]) -> List[Any]:
+    """
```
Comment (Mattwmaster58) [inline pyppeteer/coverage.py:320]:
```suggestion
    # todo: typeddict for this
    """
```

**Review by Mattwmaster58 — CHANGES_REQUESTED** [review]

File: pyppeteer/coverage.py (hunk 2 above)
Comment (Granitosaurus) [reply pyppeteer/coverage.py:164]: yes the scheduling is ordered though execution of course will not be ordered but in this case I don't think the order matters. I've been just following js code and `Promise.all` seems to behave just like `asyncio.gather` so I think we're good to go here.

File: pyppeteer/coverage.py (hunk 1 above)
Comment (Mattwmaster58) [inline pyppeteer/coverage.py]:
```suggestion
```

**Review by Mattwmaster58 — APPROVED** [review]

**Pull request merged**